internal/charmstore/store.go

```diff
@@ -566,6 +561,33 @@ func (s *Store) AddCharm(c charm.Charm, p AddParams) (err error) {
 	return nil
 }
 
+// denormalizeEntity sets all denormalized fields in e
+// from their associated canonical fields.
+//
+// It is the responsibility of the caller to set e.SupportedSeries
+// if the entity URL does not contain a series.
```

Comment:
Maybe document that e.SupportedSeries is overwritten with e.URL.Series even if e.SupportSeries is not default?